### PR TITLE
CRS-2066 - Adding Offence code validation before calling calculate

### PR DESCRIPTION
**Pull request by FaizJD030**
https://dsdmoj.atlassian.net/browse/CRS-2066

Corresponding frontend update :
https://github.com/ministryofjustice/calculate-release-dates/pull/764

### CODE REVIEW

**Review by srowlands-moj — APPROVED** [review]
LGTM only minor comment

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationCode.kt [hunk 1]
```diff
@@ -70,4 +71,9 @@ enum class ValidationCode(val message: String, val validationType: ValidationTyp
   PRE_PCSC_DTO_WITH_ADJUSTMENT("If a Detention and training order (DTO) has a sentence date before 28 June 2022, %s cannot be applied."),
   BOTUS_CONSECUTIVE_OR_CONCURRENT_TO_OTHER_SENTENCE("BOTUS concurrent/consecutive with other sentence type", UNSUPPORTED_CALCULATION),
   SDS_EARLY_RELEASE_UNSUPPORTED("SDS40 Early release is not currently supported", UNSUPPORTED_CALCULATION),
+  UNSUPPORTED_OFFENCE_ENCOURAGING_OR_ASSISTING(
+    "Any offences that include the inchoate ‘Encouraging/ Assisting’ should be recorded as the underlying act, ending in the letter ‘e’.\n" +
+      "For example, ‘Encouraging/Assisting a Rape SX03001 would be SX03001E.",
```
Comment (srowlands-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationCode.kt:76]: Minor typo missing single quote

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationService.kt [hunk 2]
```diff
@@ -307,6 +314,24 @@ class ValidationService(
...
   }
 
+  private fun findUnsupportedEncouragingOffenceCodes(sentenceAndOffences: List<SentenceAndOffence>): List<SentenceAndOffence> {
+    val offenceCodesToFilter = (2..13).map { "SC070${"%02d".format(it)}" }
+    return sentenceAndOffences.filter { it.offence.offenceCode in offenceCodesToFilter }
+  }
+
+  private fun validateUnsupportedOffences(sentencesAndOffence: List<SentenceAndOffenceWithReleaseArrangements>): List<ValidationMessage> {
+    val messages = validateUnsupportedEncouragingOffences(sentencesAndOffence).toMutableList()
+    return messages
```
[the remark this answers is not in the document]
Comment (FaizJD030) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationService.kt]: Amended as suggested

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationCode.kt (hunk 1 above)
Comment (FaizJD030) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/ValidationCode.kt]: Corrected

**Review by srowlands-moj — APPROVED** [review]
LGTM

**Pull request merged**